R/utils.R

```diff
@@ -16,8 +16,27 @@ get_first_name <- function(df){
 }
 
 #' Generate a random ID.
+#'
 #' @return A characters corresponding to random ID.
 generate_id <- function () {
   paste0(paste0(sample(c(LETTERS, letters, 0:9), 5, TRUE),
                 collapse = ""), round(as.numeric(Sys.time()) * 1000))
 }
+
+#' Detect type of predicate
+#'
+#' @param predicate Predicate or predicate generator function.
+#' @return String informing about predicate type.
+#'   When "direct" the predicate is directly applied to the column.
+#'   When "generator" the predicate returns predicate computed based on a column.
+#'   When NA, the predicate shouldn't be used at all in the validation.
+get_predicate_type <- function(predicate) {
+  predicate_output <- predicate(0:1)
+  if (is.logical(predicate_output)) {
+    return("direct")
+  }
+  if (is.function(predicate_output)) {
```

Comment:
Can we also check if `is.logical(predicate_output(0:1))`?

Reply:
done